service/serializer/src/test/java/org/jolokia/service/serializer/json/ObjectToJsonConverterTest.java

```diff
@@ -186,6 +186,16 @@ public void convertTransientValue() throws AttributeNotFoundException {
         }
     }
 
+    @Test
+    public void convertLong() throws AttributeNotFoundException {
+        long value = 900719925474099123L;
+        Object ret1 = converter.serialize(value, null, SerializeOptions.DEFAULT);
+        assertEquals(900719925474099123L, ret1);
+        SerializeOptions.Builder builder = new SerializeOptions.Builder();
+        Object ret2 = converter.serialize(value, null, builder.serializeLong("string").build());
+        assertEquals("900719925474099123", ret2);
```

Comment:
So at server side there's no place where `n` suffix is added?

Reply:
Yes. At least so long as we use json-simple, if we insert `12345n` in JSON the last step of ejecting the response to the client should fail throwing a parser exception at the server side.

Also, even if we use JSON-java that supports the bigint notation, we would then no longer align with the JSON standard as there's no mention of the bigint notation for number in its original spec.

Reply:
So please remind me - currently jquery with `$.ajax` + `ajaxParams.dataType = 'json'` simply does `JSON.parse()` and there's no way to handle `n` suffix at all?
So when long is serialized into string, we should simply rely on type coercion when someone does something like `response.value.committed + 0`?

Reply:
Yes, you are right. When we set the new option `serializeLong=string` then all long values are serialized into string, and it's all the responsibility of the client side to deserialize them into long or number with the user's prefered method.

But I'd say `serializeLong=string` will be never used with clients other than JS, as they don't have the issue with long values unlike JS.